lib/model/autocomplete.dart

```diff
@@ -204,14 +267,23 @@ class MentionAutocompleteView extends ChangeNotifier {
     final aLatestMessageId = recentDms.latestMessagesByRecipient[userA.userId];
     final bLatestMessageId = recentDms.latestMessagesByRecipient[userB.userId];
 
-    return switch((aLatestMessageId, bLatestMessageId)) {
-      (int a, int b) => -a.compareTo(b),
-      (int(),     _) => -1,
-      (_,     int()) => 1,
-      _              => 0,
-    };
+    return -compareNullable(aLatestMessageId, bLatestMessageId);
   }
 
+  /// Compares [a] to [b].
+  ///
+  /// If both are non-null, returns [a.compareTo(b)].
+  /// If [a] is null and [b] is non-null, returns a negative number.
+  /// If [a] is non-null and [b] is null, returns a positive number.
+  /// If both are null, returns zero.
+  @visibleForTesting
+  static int compareNullable(int? a, int? b) => switch ((a, b)) {
```

Comment:
From the name of this, I can't tell whether null comes before all the integers, or after all of them, or something else (equal to zero?). That makes it hard to tell whether or why this function is appropriate to use in any given context.

The name should be something that indicates that, and the summary line in the doc should indicate it too.

It could be something direct about the behavior: like `compareNullBeforeInt`, or `compareNullAsLeast`, or something like that.

Or maybe better is to give a name that reflects why it is that two methods on this class both want it: the point is that these aren't just any ints, but specifically are the greatest message IDs in two sets. When the set of message IDs is empty, its value is mathematically $-\infty$, negative infinity; we're representing that as null but it should therefore be treated as more negative than any number.

So the name could be `compareRecentMessageIds`. The doc can say:
```dart
/// Compares [a] to [b], with null less than all integers.
///
/// The values should represent the most recent message ID in each of two
/// sets of messages, with null meaning the set is empty.
///
/// Return values are as with [Comparable.compareTo].
```

And that's it — the summary line, plus the reference to [Comparable.compareTo] to resolve any ambiguity about what "compare" means, completely specifies the behavior so there's no need to transcribe the implementation into English.